test/e2e/daemonsetjobs.go

```diff
@@ -0,0 +1,107 @@
+				return err
+			}
+			return nil
+		}, "40s", "1s").ShouldNot(HaveOccurred())
+
+		Eventually(framework.HOCheckDaemonSetJobRegistration(*apiextensionsclientset), "10s", "2s").ShouldNot(HaveOccurred())
+	})
+
+	It("should create a daemonsetjob", func() {
+		workflowClient, kubeClient := framework.BuildAndSetClients()
+		ns := api.NamespaceDefault
+		myDaemonSetJob := framework.NewDaemonSetJob(workflow.GroupName, "v1", "daemonsetjob1", ns, nil)
+		defer func() {
+			deleteDaemonSetJob(workflowClient, myDaemonSetJob)
+			deleteAllJobsFromDaemonSetJob(kubeClient, myDaemonSetJob)
+		}()
+
+		Eventually(framework.HOCreateDaemonSetJob(workflowClient, myDaemonSetJob), "5s", "1s").ShouldNot(HaveOccurred())
+
+		Eventually(framework.HOIsDaemonSetJobStarted(workflowClient, myDaemonSetJob), "40s", "5s").ShouldNot(HaveOccurred())
+	})
+
+	It("should run to finish a daemonsetjob", func() {
+		workflowClient, kubeClient := framework.BuildAndSetClients()
+		ns := api.NamespaceDefault
+		myDaemonSetJob := framework.NewDaemonSetJob(workflow.GroupName, "v1", "daemonsetjob2", ns, nil)
+		defer func() {
+			deleteDaemonSetJob(workflowClient, myDaemonSetJob)
+			deleteAllJobsFromDaemonSetJob(kubeClient, myDaemonSetJob)
+		}()
+		Eventually(framework.HOCreateDaemonSetJob(workflowClient, myDaemonSetJob), "5s", "1s").ShouldNot(HaveOccurred())
+
+		Eventually(framework.HOIsDaemonSetJobFinished(workflowClient, myDaemonSetJob), "40s", "5s").ShouldNot(HaveOccurred())
+
+		Eventually(framework.HOCheckAllDaemonSetJobJobsFinished(workflowClient, myDaemonSetJob), "60s", "5s").ShouldNot(HaveOccurred())
+	})
+
+	It("should not start job since NodeSelector dont match", func() {
+		workflowClient, kubeClient := framework.BuildAndSetClients()
+		ns := api.NamespaceDefault
+		myDaemonSetJob := framework.NewDaemonSetJob(workflow.GroupName, "v1", "daemonsetjob3", ns, map[string]string{"sdfdsfsffsd": "fsdfsdfsdfds"})
+		defer func() {
+			deleteDaemonSetJob(workflowClient, myDaemonSetJob)
+			deleteAllJobsFromDaemonSetJob(kubeClient, myDaemonSetJob)
+		}()
+		Eventually(framework.HOCreateDaemonSetJob(workflowClient, myDaemonSetJob), "5s", "1s").ShouldNot(HaveOccurred())
+
+		Eventually(framework.HOIsDaemonSetJobStarted(workflowClient, myDaemonSetJob), "40s", "5s").ShouldNot(HaveOccurred())
+
+		Eventually(framework.HOCheckZeroDaemonSetJobJobsWasCreated(workflowClient, myDaemonSetJob), "10s", "5s").ShouldNot(HaveOccurred())
+	})
+
+})
+
+func deleteDaemonSetJob(workflowClient versioned.Interface, daemonsetjob *dapi.DaemonSetJob) {
+	workflowClient.DaemonsetjobV1().DaemonSetJobs(daemonsetjob.Namespace).Delete(daemonsetjob.Name, nil)
+	By("DaemonSetJob deleted")
+}
+
+func deleteAllJobsFromDaemonSetJob(kubeClient clientset.Interface, daemonsetjob *dapi.DaemonSetJob) {
+	jobs, err := kubeClient.Batch().Jobs(daemonsetjob.Namespace).List(metav1.ListOptions{
+		LabelSelector: controller.DaemonSetJobLabelKey + "=" + daemonsetjob.Name,
```

Comment:
just curious. We don't have a function that instantiate the proper label?

Reply:
I should use some apimachinery struct, like labels.Set and then convert it to a String